solr/core/src/test/org/apache/solr/TestCpuTimeSearch.java

```diff
@@ -0,0 +1,140 @@
+    shard2 = urlCollection2.replaceAll("https?://", "");
+
+    SolrInputDocument doc = new SolrInputDocument();
+    doc.setField("id", "1");
+    doc.setField("subject", "batman");
+    doc.setField("title", "foo bar");
+    collection1.add(doc);
+    collection1.commit();
+
+    doc.setField("id", "2");
+    doc.setField("subject", "superman");
+    collection2.add(doc);
+    collection2.commit();
+  }
+
+  @AfterClass
+  public static void cleanup() throws Exception {
+    if (null != collection1) {
+      collection1.close();
+      collection1 = null;
+    }
+    if (null != collection2) {
+      collection2.close();
+      collection2 = null;
+    }
+
+    System.setProperty(ThreadStats.ENABLE_CPU_TIME, "false");
+    System.setProperty(AllowListUrlChecker.DISABLE_URL_ALLOW_LIST, "false");
+  }
+
+  public void testWithoutDistrib() throws SolrServerException, IOException {
+    SolrQuery query = new SolrQuery();
+    query.setQuery("subject:batman OR subject:superman");
+    query.addField("id");
+    query.addField("subject");
+
+    QueryResponse response = collection1.query(query);
+
+    SolrDocumentList results = response.getResults();
+    int size = results.size();
+    assertEquals("should have 1 result", 1, size);
+    NamedList<Object> header = response.getHeader();
+    List<Object> localCpuTimes = header.getAll(ThreadStats.LOCAL_CPU_TIME);
+    List<Object> cpuTimes = header.getAll(ThreadStats.CPU_TIME);
+    assertEquals("localCpuTime should not have values", 0, localCpuTimes.size());
+    assertEquals("cpuTime should only have one value", 1, cpuTimes.size());
+    long cpuTime = (long) cpuTimes.iterator().next();
+    assertTrue("cpuTime (" + cpuTime + ") should be positive", cpuTime >= 0);
+  }
+
+  public void testWithDistrib() throws SolrServerException, IOException {
+    SolrQuery query = new SolrQuery();
+    query.setQuery("subject:batman OR subject:superman");
+    query.addField("id");
+    query.addField("subject");
+    query.set("distrib", "true");
+    query.set("shards", shard1 + "," + shard2);
+    query.set(ShardParams.SHARDS_INFO, "true");
+    query.set("debug", "true");
+    query.set("stats", "true");
+    query.set("stats.field", "id");
+    query.set(ShardParams.SHARDS_TOLERANT, "true");
+
+    QueryResponse response = collection1.query(query);
+
+    SolrDocumentList results = response.getResults();
+    int size = results.size();
+    assertEquals("should have 2 results", 2, size);
+    NamedList<Object> header = response.getHeader();
+    List<Object> localCpuTimes = header.getAll(ThreadStats.LOCAL_CPU_TIME);
+    List<Object> cpuTimes = header.getAll(ThreadStats.CPU_TIME);
+    assertEquals("localCpuTime should not have values", 0, localCpuTimes.size());
+    assertEquals("cpuTime should only have one value", 1, cpuTimes.size());
+    long cpuTime = (long) cpuTimes.iterator().next();
+    assertTrue("cpuTime (" + cpuTime + ") should be positive", cpuTime >= 0);
+  }
```

Comment:
we could also assert that cpuTimes >= sum(localCpuTimes)

Reply:
only cpuTimes is returned so it is not possible to make this check from the client time.
We decided earlier that it was better not to let the client knows if the query was distributed